WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1882,6 +1874,66 @@ extension ReaderStreamViewController: ReaderPostUndoCellDelegate {
 }
 
 
+// MARK: - Select Interests Display
+private extension ReaderStreamViewController {
+    func displaySelectInterestsIfNeeded(_ content: ReaderContent) {
+        guard
+            FeatureFlag.readerImprovementsPhase2.enabled,
+            content.topicType == .discover
+        else {
+            // Hide the view if we're not on the discover tab, and it exists
+            self.selectInterestsViewController?.remove()
+            return
+        }
+
+        if self.selectInterestsViewController != nil {
+            self.showSelectInterestsViewIfNeeded()
+            return
+        }
+
+        // If we're not showing the select interests view, check to see if we should
+        interestsCoordinator.shouldDisplay { [unowned self] shouldDisplay in
+            if shouldDisplay {
+                self.makeSelectInterestsViewControllerIfNeeded()
+                self.showSelectInterestsViewIfNeeded()
+            }
+        }
+    }
+
+    func showSelectInterestsViewIfNeeded() {
+        guard let controller = self.selectInterestsViewController else {
+            return
+        }
+
+        UIView.animate(withDuration: 0) {
+            controller.view.frame = self.view.bounds
+            self.add(controller, asChildOf: self)
+        }
+    }
+
+    func makeSelectInterestsViewControllerIfNeeded() {
+        if selectInterestsViewController != nil {
+            return
+        }
+
+        let controller = ReaderSelectInterestsViewController()
+        controller.didSaveInterests = { [unowned self] in
+            guard let controller = self.selectInterestsViewController else {
```

Comment:
But wouldn't selectInterestsViewController be empty? Isn't that what we're checking at the beginning?